packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -183,6 +186,41 @@ function ListViewBlockSelectButton(
 					updateFocusAndSelection( updatedBlocks[ 0 ], false );
 				}
 			}
+		} else if ( isMatch( 'core/block-editor/select-all', event ) ) {
+			if ( event.defaultPrevented ) {
+				return;
+			}
+			event.preventDefault();
+
+			const { firstBlockRootClientId, selectedBlockClientIds } =
+				getBlocksToUpdate();
+			const blockClientIds = getBlockOrder( firstBlockRootClientId );
+			if ( ! blockClientIds.length ) {
+				return;
+			}
+
+			// If we have selected all sibling nested blocks, try selecting up a level.
+			// This is a similar implementation to that used by `useSelectAll`.
+			if ( selectedBlockClientIds.length === blockClientIds.length ) {
```

Comment:
This logic isn't accurate enough in the list view because the list view can be focused in a different part of the hierarchy than the selection, so it's possible for the number of siblings of the currently focused block to match the length of the selection elsewhere in the list.

I think we'll need to replace this with an exact match to ensure ids in both lists are the same.